File: controllers/task_controller.py
from flask import request, jsonify
from models.task_model import task_model

def list_tasks():
    tasks = task_model.get_all()
    return jsonify(tasks), 200

def get_task(task_id):
    task = task_model.get_by_id(task_id)
    if not task:
        return jsonify({'error': 'Task not found'}), 404
    return jsonify(task), 200

def create_task():
    data = request.get_json() or {}
    title = data.get('title')
    description = data.get('description', '')
    if not title or not isinstance(title, str):
        return jsonify({'error': 'title is required and must be a string'}), 400
    task = task_model.create(title=title, description=description)
    return jsonify(task), 201

def update_task(task_id):
    data = request.get_json() or {}
    # validate minimalement completed type si présent
    if 'completed' in data and not isinstance(data['completed'], bool):
        return jsonify({'error': 'completed must be a boolean'}), 400
    task = task_model.update(task_id, data)
    if not task:
        return jsonify({'error': 'Task not found'}), 404
    return jsonify(task), 200

def delete_task(task_id):
    ok = task_model.delete(task_id)
    if not ok:
        return jsonify({'error': 'Task not found'}), 404
    return '', 204

File: controllers/task_controller.js
// controllers/taskController.js
const taskModel = require('../models/taskModel');

exports.listTasks = (req, res) => {
  const tasks = taskModel.getAll();
  res.json(tasks);
};

exports.getTask = (req, res) => {
  const { id } = req.params;
  const task = taskModel.getById(id);
  if (!task) return res.status(404).json({ error: 'Task not found' });
  res.json(task);
};

exports.createTask = (req, res) => {
  try {
    const { title, description } = req.body;
    const task = taskModel.create({ title, description });
    res.status(201).json(task);
  } catch (err) {
    res.status(400).json({ error: err.message });
  }
};

exports.updateTask = (req, res) => {
  const { id } = req.params;
  const fields = req.body;
  // optional: validate completed type if present
  if ('completed' in fields && typeof fields.completed !== 'boolean') {
    return res.status(400).json({ error: 'completed must be a boolean' });
  }
  const updated = taskModel.update(id, fields);
  if (!updated) return res.status(404).json({ error: 'Task not found' });
  res.json(updated);
};

exports.deleteTask = (req, res) => {
  const { id } = req.params;
  const ok = taskModel.delete(id);
  if (!ok) return res.status(404).json({ error: 'Task not found' });
  res.status(204).send();
};
